OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -137,7 +110,7 @@ public int ContentHash
 		}
 
 		public Harvester(Actor self, HarvesterInfo info)
-			: base(info)
+			: base(self, info)
 		{
 			Contents = new ReadOnlyDictionary<string, int>(contents);
```

Comment:
Later in the code you refer to `Contents` as Resources. So it would be better to either rename `Contents` to `Resources`, or change the later `resourceType` var into `contentsType`, to be consistent and avoid any confusion.

Reply:
I think it would be confusing. Contents - something that is contained / stored. In this case Resources

Anyway that has nothing to do with this refactor

Reply:
Ok I think this could be its own PR, but just to entertain the point further: In this case I think it should be `contentsType`, because contents can mean anything, yet it is being treated only as resources in the below comment. I actually prefer `contentsType`, because it opens the door for non-resource related contents to be stored, though if you are doing this you would need an enum specifying which contentsTypes are resourceTypes, so you can then check for this first before handling it with `acceptResources`. If you really wanted you could put a comment stating: "NOTE: All harvester contents are assumed to be resources." or similar, and just leave `resourceType = c.Key` as is, though the former approach would be nicer. Anyway I only mention this to raise the issue with the current implementation and provide a potential solution, even if it's not the point of this PR.